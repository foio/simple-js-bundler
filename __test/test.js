const path = require('path');
const parseDependencies = require('../bundle').parseDependencies;
const getDenpendenciesGraph = require('../bundle').getDenpendenciesGraph;
const bundle = require('../bundle').bundle;

test('test parse file dependencies', () => {
    const dependenciesObj = parseDependencies('./__test/code/index.js');
    expect(dependenciesObj).toMatchSnapshot();
});

test('test parse file garph', () =>{
    const dependenciesGraph = getDenpendenciesGraph('./__test/code/index.js');
    expect(dependenciesGraph).toMatchSnapshot();
});

test('generate bundle file from entry', ()=>{
    const bundleDisFile = bundle('./__test/code/index.js');
    expect(bundleDisFile).toMatchSnapshot();
})