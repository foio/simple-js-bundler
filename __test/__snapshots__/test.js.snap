// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate bundle file from entry 1`] = `
"
        (function(graph) {
            //require函数的本质是执行一个模块的代码，然后将相应变量挂载到exports对象上
            function require(module) {
                //localRequire的本质是拿到依赖包的exports变量
                function localRequire(relativePath) {
                    return require(graph[module].dependencies[relativePath]);
                }
                var exports = {};
                (function(require, exports, code) {
                    eval(code);
                })(localRequire, exports, graph[module].code);
                return exports;
            }
            require('./__test/code/index.js')
        })({\\"./__test/code/index.js\\":{\\"code\\":\\"\\\\\\"use strict\\\\\\";\\\\n\\\\nvar _message = _interopRequireDefault(require(\\\\\\"./message.js\\\\\\"));\\\\n\\\\nfunction _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\\\\\"default\\\\\\": obj }; }\\\\n\\\\nconsole.log(_message[\\\\\\"default\\\\\\"]);\\",\\"dependencies\\":{\\"./message.js\\":\\"./__test\\\\\\\\code\\\\\\\\message.js\\"}},\\"./__test\\\\\\\\code\\\\\\\\message.js\\":{\\"code\\":\\"\\\\\\"use strict\\\\\\";\\\\n\\\\nObject.defineProperty(exports, \\\\\\"__esModule\\\\\\", {\\\\n  value: true\\\\n});\\\\nexports[\\\\\\"default\\\\\\"] = void 0;\\\\n\\\\nvar _word = require(\\\\\\"./word.js\\\\\\");\\\\n\\\\nvar message = \\\\\\"say \\\\\\".concat(_word.word);\\\\nvar _default = message;\\\\nexports[\\\\\\"default\\\\\\"] = _default;\\",\\"dependencies\\":{\\"./word.js\\":\\"./__test\\\\\\\\code\\\\\\\\word.js\\"}},\\"./__test\\\\\\\\code\\\\\\\\word.js\\":{\\"code\\":\\"\\\\\\"use strict\\\\\\";\\\\n\\\\nObject.defineProperty(exports, \\\\\\"__esModule\\\\\\", {\\\\n  value: true\\\\n});\\\\nexports.word = void 0;\\\\nvar word = 'hello';\\\\nexports.word = word;\\",\\"dependencies\\":{}}})"
`;

exports[`test parse file dependencies 1`] = `
Object {
  "code": "\\"use strict\\";

var _message = _interopRequireDefault(require(\\"./message.js\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

console.log(_message[\\"default\\"]);",
  "dependencies": Object {
    "./message.js": "./__test\\\\code\\\\message.js",
  },
  "filename": "./__test/code/index.js",
}
`;

exports[`test parse file garph 1`] = `
Object {
  "./__test/code/index.js": Object {
    "code": "\\"use strict\\";

var _message = _interopRequireDefault(require(\\"./message.js\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

console.log(_message[\\"default\\"]);",
    "dependencies": Object {
      "./message.js": "./__test\\\\code\\\\message.js",
    },
  },
  "./__test\\\\code\\\\message.js": Object {
    "code": "\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;

var _word = require(\\"./word.js\\");

var message = \\"say \\".concat(_word.word);
var _default = message;
exports[\\"default\\"] = _default;",
    "dependencies": Object {
      "./word.js": "./__test\\\\code\\\\word.js",
    },
  },
  "./__test\\\\code\\\\word.js": Object {
    "code": "\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.word = void 0;
var word = 'hello';
exports.word = word;",
    "dependencies": Object {},
  },
}
`;
